### PR TITLE
Environment variable for heartbeat interval

### PR DESCRIPTION
**Pull request by kusigit**
The heartbeat interval should be configurable via an environment variable. E.g. the request timeout on the Heroku Cloud is 55 seconds.

https://devcenter.heroku.com/articles/request-timeout#long-polling-and-streaming-responses

The implementation is just for the single process. Maybe we have to adapt it to microservices as well.

### CODE REVIEW

**Comment by goloroden** [comment]
Hey @kusigit 👋 

Thanks a lot for your PR!
We will have a closer look at it next week, and then get back to you.

Have a nice weekend,


Golo

**Comment by goloroden** [comment]
@kusigit So, as promised, we get back to you this week 😊

The PR looks pretty good, however, we have two questions on it:

1. Woud you like to add yourself as a contributor to the `package.json` file?
2. Would you mind adding the same settings to the micro-service runtime?

Cheers,


Golo

File: lib/runtimes/microservice/processes/notification/configurationDefinition.ts [hunk 1]
```diff
@@ -26,6 +26,11 @@ const configurationDefinition: ConfigurationDefinition<Configuration> = {
     defaultValue: 3_001,
     schema: portOrSocketSchema
   },
+  heartbeatInterval: {
+    environmentVariable: 'HEARTBEAT_INTERVAL',
+    defaultValue: 90_000,
+    schema: { type: 'integer' }
```
Comment (goloroden) [inline lib/runtimes/microservice/processes/notification/configurationDefinition.ts:32]: Should we maybe extract this to a `heartbeatIntervalSchema`, similar to how we did it with ports?

And: Shouldn't we set a `minimum` in that schema, then?

(The same applies to the other places where this parameter is defined.)

**Review by goloroden — CHANGES_REQUESTED** [review]
Hey @kusigit 👋 

Thanks for the quick update, this looks pretty good IMHO 😊 

There is one last thing (see my comment). Would you mind to have another look?

**Review by goloroden — APPROVED** [review]

**Comment by goloroden** [comment]
@kusigit Thanks for the quick update 😊

@yeldiRium To me, this PR looks fine – can you please have another look at it, and (if you are fine with it), merge it?

File: lib/runtimes/microservice/processes/notification/configurationDefinition.ts (hunk 1 above)
Comment (kusigit) [reply lib/runtimes/microservice/processes/notification/configurationDefinition.ts]: Hi @goloroden 
You're right. Good idea. Thanks.

**Review by yeldiRium — APPROVED** [review]

**Comment by yeldiRium** [comment]
LGTM :+1:

**Pull request merged**